Patterns.h

```diff
@@ -17,6 +17,33 @@
 #define PATTERNS_ENABLE_EXCEPTIONS
 #endif
 
+// This is inspired by the char_traits<unsigned char> implementation from nlohmann's json library
+struct pattern_traits : std::char_traits<char>
```

Comment:
Have you tried putting this inside `hook::basic_pattern` as a struct private to that class? Or protected? Or if that doesn't work, at least in the `hook` namespace?

Reply:
I put that struct inside the `hook` namespace for now (I had compile errors when putting it into the `basic_pattern` class)